engine/src/main/java/org/destinationsol/game/ObjectManager.java

```diff
@@ -126,8 +127,8 @@ public void update(SolGame game) {
             }
             if (isNear(fod, camPos, ts)) {
                 SolObject o = fo.toObj(game);
-                // Ensure that StarPorts are added straight away so that we can see if they overlap
-                if (o instanceof StarPort) {
+                // Ensure that StarPorts and Asteroids are added straight away so that we can see if they overlap
+                if (o instanceof StarPort || o instanceof FarAsteroid) {
```

Comment:
With addObjectNow(), you are adding objects immediately into the current frame of the game; if you for instance added object if the half of the execution of current frames update(), half of the methods that are executed each frame wouldn't be performed while aware of the object. If some of these methods, for instance, set up these objects, for instance some of their inner variables, like shouldBeRemoved (not the greatest example), the object, inserted with addObjectNow would be removed a frame or two later. With addObjectsDelayed, however, you ensure that the objects will be inserted right before the beginning of the next frame, and thus everything that has to be executed in the frame will be aware of these objects and it may take them in account.